homeassistant/components/switch/flux.py

```diff
@@ -0,0 +1,189 @@
+"""
+Flux for Home-Assistant.
+
+The idea was taken from https://github.com/KpaBap/hue-flux/
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch/flux/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.light import is_on, turn_on
+from homeassistant.components.sun import next_setting, next_rising
+from homeassistant.components.switch import DOMAIN, SwitchDevice
+from homeassistant.const import CONF_NAME, CONF_PLATFORM, EVENT_TIME_CHANGED
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util.color import color_temperature_to_rgb as temp_to_rgb
+from homeassistant.util.color import color_RGB_to_xy
+from homeassistant.util.dt import now as dt_now
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['sun', 'light']
+SUN = "sun.sun"
+_LOGGER = logging.getLogger(__name__)
+
+CONF_LIGHTS = 'lights'
+CONF_START_TIME = 'start_time'
+CONF_STOP_TIME = 'stop_time'
+CONF_START_CT = 'start_colortemp'
+CONF_SUNSET_CT = 'sunset_colortemp'
+CONF_STOP_CT = 'stop_colortemp'
+CONF_BRIGHTNESS = 'brightness'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'flux',
+    vol.Required(CONF_LIGHTS): cv.entity_ids,
+    vol.Optional(CONF_NAME, default="Flux"): cv.string,
+    vol.Optional(CONF_START_TIME, default=None): cv.time,
+    vol.Optional(CONF_STOP_TIME,
+                 default=dt_now().replace(hour=22, minute=0)): cv.time,
+    vol.Optional(CONF_START_CT, default=4000):
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_SUNSET_CT, default=3000):
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_STOP_CT, default=1900):
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_BRIGHTNESS, default=None):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255))
+})
+
+
+def set_lights_xy(hass, lights, x_val, y_val, brightness):
+    """Set color of array of lights."""
+    for light in lights:
+        if is_on(hass, light):
+            turn_on(hass, light,
+                    xy_color=[x_val, y_val],
+                    brightness=brightness,
+                    transition=30)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the demo switches."""
+    name = config[CONF_NAME]
+    lights = config[CONF_LIGHTS]
+    start_time = config.get(CONF_START_TIME)
+    stop_time = config.get(CONF_STOP_TIME) or dt_now().replace(hour=22, minute=0)
```

Comment:
I don't know why I had to add the `or` here for defaults.  Shouldn't the defaults in the `PLATFORM_SCHEMA` above take care of it?